src/test/metric_config_test.cpp

```diff
@@ -0,0 +1,257 @@
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ],
+    "monitoring":
+        {
+            "metrics":
+            {
+                "enable" : true,
+                "endpoint_path": "/new..metrics"
+            }
+        }
+})";
+
+class MetricsConfigTest : public TestWithTempDir {
+public:
+    std::string configFilePath;
+    std::string ovmsConfig;
+    std::string modelPath;
+    std::string dummyModelName;
+
+    void SetUpConfig(const std::string& configContent) {
+        ovmsConfig = configContent;
+        dummyModelName = "dummy";
+        const std::string modelPathToReplace{"/ovms/src/test/dummy"};
+        ovmsConfig.replace(ovmsConfig.find(modelPathToReplace), modelPathToReplace.size(), modelPath);
+    }
+    void SetUp() override {
+        TestWithTempDir::SetUp();
+        // Prepare manager
+        modelPath = directoryPath + "/dummy/";
+        configFilePath = directoryPath + "/ovms_config.json";
+    }
+};
+
+TEST_F(MetricsConfigTest, DefaultValues) {
+    SetUpConfig(modelDefaultConfig);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+
+    const auto& metricConfig = manager.getMetricConfig();
+    ASSERT_EQ(metricConfig.metricsEnabled, false);
+    ASSERT_EQ(metricConfig.endpointsPath, "/metrics");
+    ASSERT_EQ(metricConfig.requestSuccessGrpcPredict, false);
+    ASSERT_EQ(metricConfig.requestFailRestModelReady, false);
+}
+
+TEST_F(MetricsConfigTest, ChangedValues) {
+    SetUpConfig(modelMetricsChangedConfig);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+
+    const auto& metricConfig = manager.getMetricConfig();
+    ASSERT_EQ(metricConfig.metricsEnabled, true);
+    ASSERT_EQ(metricConfig.endpointsPath, "/newmetrics");
+    ASSERT_EQ(metricConfig.requestSuccessGrpcPredict, true);
+    ASSERT_EQ(metricConfig.requestFailRestModelReady, true);
+    ASSERT_EQ(metricConfig.requestSuccessGrpcModelReady, false);
+}
+
+TEST_F(MetricsConfigTest, InitOnce) {
+    SetUpConfig(modelMetricsChangedConfig);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+    TearDown();
```

Comment:
Why do we need to tear down and re-setup?
If I'm not mistaken, `InitOnce` has intention to test if metric config does *not* change after second `loadConfig` call. If so, why can't we just call loadConfig second time and check values?

Reply:
TearDown is for new file to be added to directory. Test cant overwrite old file - there was an error. Objects do not change.